cylc/flow/templatevars.py

```diff
@@ -20,17 +20,18 @@
 from typing import Any, Dict
 
 from cylc.flow.exceptions import InputError
-
-
 from cylc.flow.rundb import CylcWorkflowDAO
+from cylc.flow.workflow_db_mgr import WorkflowDatabaseManager
 
 
 def get_template_vars_from_db(run_dir):
     """Get template vars stored in a workflow run database.
     """
     template_vars = {}
     if (run_dir / 'log/db').exists():
-        dao = CylcWorkflowDAO(str(run_dir / 'log/db'))
+        WorkflowDatabaseManager(
+            str(run_dir / 'log')).check_workflow_db_compatibility()
```

Comment:
Should this be using the private DB?

Reply:
There is a problem with this - `WorkflowDatabaseManager().check_workflow_db_compatibility()` will try to create the tables. I have just realised I had commented this in the code in a "tidy" branch but never created a PR

Reply:
I think:

* We need to add a switch to allow `check_workflow_db_compatibility` to run on the public DB?
  * Or are we really really sure the workflow is not running at this point?
* We should avoid that `CREATE TABLES` junk for this as it's just another [unnecessary] thing to go wrong.

Reply:
I've made some changes which ought to fix this - 
Essentially allowing the `check_workflow_db_compatibility` method to specify whether use public or private databases.

Reply:
Resolved by #5330